src/main/java/org/embl/mobie/io/SpimDataOpener.java

```diff
@@ -156,18 +157,40 @@ private SpimData openOpenOrganelleS3(String path) throws SpimDataException {
         }
     }
 
-    private SpimData openBdvOmeZarrS3(String path, SharedQueue queue) throws SpimDataException {
+    private SpimData openBdvOmeZarrS3(String path, SharedQueue queue) {
+        //Todo: finish bug fixing
         try {
-            N5S3OMEZarrImageLoader imageLoader = createN5S3OmeZarrImageLoader(path, queue);
-            SpimData spimData = openBdvXml(path);
-            if (spimData != null) {
-                spimData.getSequenceDescription().setImgLoader(imageLoader);
-                return spimData;
+            SAXBuilder sax = new SAXBuilder();
+            InputStream stream = FileAndUrlUtils.getInputStream(path);
+            Document doc = sax.build(stream);
+            Element imgLoaderElem = doc.getRootElement().getChild("SequenceDescription").getChild("ImageLoader");
+            String bucketAndObject = imgLoaderElem.getChild("BucketName").getText() + "/" + imgLoaderElem.getChild("Key").getText();
+            String[] split = bucketAndObject.split("/");
+            String bucket = split[0];
+            String object = Arrays.stream(split).skip(1L).collect(Collectors.joining("/"));
+            N5S3OMEZarrImageLoader imageLoader;
+            if (queue != null) {
+                imageLoader = new N5S3OMEZarrImageLoader(imgLoaderElem.getChild("ServiceEndpoint").getText(), imgLoaderElem.getChild("SigningRegion").getText(), bucket, object, ".", queue);
             } else {
-                throw new SpimDataException("Error while trying to read spimData. SpimData is null");
+               imageLoader = new N5S3OMEZarrImageLoader(imgLoaderElem.getChild("ServiceEndpoint").getText(), imgLoaderElem.getChild("SigningRegion").getText(), bucket, object, ".");
             }
-        } catch (IOException | JDOMException e) {
-            throw new SpimDataException(ERROR_WHILE_TRYING_TO_READ_SPIM_DATA + e.getMessage());
+            SpimData spim = new SpimData(null, Cast.unchecked(imageLoader.getSequenceDescription()), imageLoader.getViewRegistrations());
+            SpimData sp1;
+            try {
+                InputStream st = FileAndUrlUtils.getInputStream(path);
+                sp1 = (new CustomXmlIoSpimData()).loadFromStream(st, path);
+            } catch (SpimDataException var3) {
+                System.out.println(path);
```

Comment:
@KateMoreva 
Could you replace `var3` by a more meaningful name?
In this case `spimDataException`.
Are you using IntelliJ? This usually recommend good variable names.
In general, names like `sp1` and `var4` make the code very hard to read, please try using variable names that explain what the content of the variable is.

Reply:
This code does not fix all problems. I'll rewrite it in the next PR.